[ibnu2651] iP

ibnu2651 opened the pull request:
### Ada
> "This is a quote." - Ibnu

Ada frees your mind of having to remember things you need to do. It's,

- text based
- ~hard~ _easy_

All you need to do is,

1. Download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/142)
2. Run it 🏃 

Also **FREE**
Features:

- [x] Managing tasks
- [x] Not much else
---
Here's the `main` method:

```java
public class Ada {
    public static void main(String[] args) {
        new Ada("ada.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ibnu2651 committed:
edit Duke.java

ibnu2651 committed:
Add echo to Duke

ibnu2651 committed:
Add list to Duke

ibnu2651 committed:
Add mark tasks to Duke

ibnu2651 committed:
Add todo, deadline, and event to Duke

ibnu2651 committed:
Add auto text ui testing

ibnu2651 committed:
Add error handling to Duke

ibnu2651 committed:
Add delete and more error handling to Duke

ibnu2651 committed:
Edit Expected.txt

ibnu2651 committed:
Add saving to Duke

ibnu2651 committed:
Merge branch 'branch-level-7'

ibnu2651 committed:
Add duke.txt to gitignore

ibnu2651 committed:
Add date and time to deadlines in Duke

ibnu2651 committed:
Merge branch 'branch-Level-8'

ibnu2651 committed:
Add parser, storage, tasklist, and ui to Duke

ibnu2651 committed:
Add more to parser, storage, tasklist, and ui

ibnu2651 committed:
Add command classes

ibnu2651 committed:
Add more OOP to Duke

ibnu2651 committed:
Merge branch 'branch-A-MoreOOP'

ibnu2651 committed:
Separate duke into packages

ibnu2651 committed:
Merge branch 'branch-A-Packages'

ibnu2651 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ibnu2651 committed:
Add gradle to duke

ibnu2651 committed:
Merge branch 'branch-A-Gradle'

ibnu2651 committed:
Add JUnit testing for deadline and tasklist

ibnu2651 committed:
Merge branch 'branch-A-JUnit'

ibnu2651 committed:
Add javadocs

ibnu2651 committed:
Tweak code to comply with coding standard

ibnu2651 committed:
Add find to duke

ibnu2651 committed:
Merge branch 'branch-Level-9'

ibnu2651 committed:
Merge branch 'branch-A-CodingStandard'

ibnu2651 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java

ibnu2651 committed:
Rename duke to ada

ibnu2651 committed:
Refactor files to use ada

ibnu2651 committed:
Rename Ada to Adam and fix some text

angsongyee commented on src/main/java/adam/Parser.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    public static Command parse(String fullCommand) throws AdaException {
+        String command = fullCommand.split(" ")[0];
+        String[] splitCommand;
+        String[] splitDesc;
```
Technically the variables here should be in plural form since they are of type `String[]` but I can't think of a suitable replacement name either😅.

angsongyee commented on src/main/java/adam/command/AddCommand.java:
```diff
@@ -0,0 +1,69 @@
+public class AddCommand extends Command {
+    private String command;
+    private String desc;
+    private String[] details;
```
Good job on following the plural naming style here 👍

angsongyee commented on src/main/java/adam/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+     * Marks the task as done.
+     */
+    public void mark() {
+        this.isDone = true;
```
Its a good practice to omit the `this` keyword when accessing class members and you should only use it when there are ambiguous variable names in the scope (i.e. in constructors or when your method arguments have the same 
name)
```suggestion
        isDone = true;
```

angsongyee reviewed (commented):
Overall, the code is quite readable, and it follows the coding standard very well, other than a view minor nitpicks. Also, I noticed that currently there is some mismatch between the names in the project (e.g. filename says Adam.java but package and class name is Ada) but maybe it's because you haven't completely refactored everything yet so its not a problem for me. 👍 

(I just realized that I sent all my review comments individually instead of combining them, sorry for spamming your inbox 🙇)

ibnu2651 committed:
Add GUI to Adam

ibnu2651 committed:
Fix DialogBox resizing

ibnu2651 committed:
Merge branch 'branch-Level-10'

ibnu2651 commented:
Thanks for the helpful comments, yeah I'm not too sure what happened with the renaming, maybe doing it at 3 am was not a good idea. Don't worry about the multiple comments thing, it's all good. 👍

ibnu2651 committed:
Change names based on PR comments

ibnu2651 committed:
Change image for Adam

ibnu2651 committed:
Add assertions to Adam

There are no verifications in the code.

Adding assertions will verify assumptions about the program state and
catch potential bugs in the code.

Let's add assertions when adding a task and performing other commands
on tasks.

ibnu2651 committed:
Improve code quality of Adam

ibnu2651 committed:
Merge pull request #1 from ibnu2651/branch-A-Assertions

Add assertions to Adam

ibnu2651 committed:
Merge branch 'master' into branch-A-CodeQuality

ibnu2651 committed:
Merge pull request #2 from ibnu2651/branch-A-CodeQuality

Improve code quality of Adam

ibnu2651 committed:
Merge branch 'master' of https://github.com/ibnu2651/ip

ibnu2651 committed:
Add shorter aliases for keywords

Command syntax is wordy and inflexible.

Shortened commands that do the same thing as the original command is
more user-friendly as they are easier and faster to type.

Let's add shorter aliases of up to 3 characters that behave
identically to the normal verbose commands they are derived from.

ibnu2651 committed:
Merge branch 'branch-C-FriendlierSyntax'

shunjieee commented on src/main/java/adam/Adam.java:
Well abstracted and naming! No striking violation to the naming conventions. :)

ibnu2651 committed:
Update adam.txt

ibnu2651 committed:
Update adam.txt

ibnu2651 committed:
Update Javadocs for most files

ibnu2651 committed:
Add .yml file into workflow

ibnu2651 committed:
Edit workflow .yml

ibnu2651 committed:
Merge branch 'branch-A-CI'

ibnu2651 committed:
Change files to match checkstyle requirements

ibnu2651 committed:
Merge branch 'branch-A-CheckStyle'

ibnu2651 committed:
Add Ui.png

ibnu2651 committed:
Update User Guide

ibnu2651 committed:
Update User Guide

ibnu2651 committed:
Refactor Parser

ibnu2651 committed:
Update JUnit tests

ibnu2651 committed:
Update User Guide

ibnu2651 committed:
Update download link in User Guide

ibnu2651 committed:
Add name to GUI

ibnu2651 committed:
Add image to User Guide

ibnu2651 committed:
Change storage and FXML version
